packages/frontend/src/ee/features/aiCopilot/components/ChatElements/ToolCalls/AiChartToolCalls.tsx

```diff
@@ -0,0 +1,147 @@
+import {
+    type AiAgentToolCall,
+    TOOL_DISPLAY_MESSAGES_AFTER_TOOL_CALL,
+    type ToolName,
+    isFindFieldsToolArgs,
+} from '@lightdash/common';
+import { Badge, Group, Stack, Text, Timeline } from '@mantine-8/core';
+import {
+    IconChartHistogram,
+    IconChartLine,
+    IconFileText,
+    IconFilter,
+    IconSearch,
+    IconSparkles,
+} from '@tabler/icons-react';
+import { type FC } from 'react';
+import MantineIcon from '../../../../../../components/common/MantineIcon';
+
+const getToolIcon = (toolName: string) => {
+    const iconMap = {
+        findFields: IconSearch,
+        generateBarVizConfig: IconChartHistogram,
+        generateTimeSeriesVizConfig: IconChartLine,
+        generateQueryFilters: IconFilter,
+        generateCsv: IconFileText,
+    } as const;
+
+    return iconMap[toolName as keyof typeof iconMap] || IconSparkles;
+};
+
+const getToolDescription = (toolCall: AiAgentToolCall) => {
+    const { toolName, toolArgs } = toolCall;
+
+    // Type guard to ensure toolArgs exists
+    if (!toolArgs) {
+        return (
+            <Text c="dimmed" size="sm">
+                {TOOL_DISPLAY_MESSAGES_AFTER_TOOL_CALL[
+                    toolCall.toolName as ToolName
+                ] || `Executed ${toolCall.toolName}`}
+            </Text>
+        );
+    }
+
+    switch (toolName) {
```

Comment:
if this is enum / union type then you can do exhaustive switch case with `assertUnreachable`